### PR TITLE
Tests for draw functions

### PR DESCRIPTION
**Pull request by maximelucas**
- tests for draw functions
- added defaults param values to all draw functions
- tests for clustering

### CODE REVIEW

File: tests/drawing/test_draw.py [hunk 1]
```diff
@@ -85,3 +87,55 @@ def test_color_arg_to_dict(edgelist4):
...
+
+    ax = xgi.draw(H)
+
+    assert len(ax.lines) == len(H.edges.filterby("size", 2))  # dyads
+    assert len(ax.patches) == len(H.edges.filterby("size", 2, mode="gt"))  # hyperedges
+    assert len(ax.collections[0].get_sizes()) == H.num_nodes  # nodes
+    plt.close()
+
+
+def test_draw_nodes(edgelist8):
```
Comment (nwlandry) [inline tests/drawing/test_draw.py:104]: Can you add a test to make sure the lines, nodes, and hyperedges/simplices have the right zorder?

File: xgi/drawing/draw.py [hunk 2]
```diff
@@ -888,3 +958,21 @@ def draw_hyperedge_labels(
         text_items[id] = t
 
     return text_items
+
+
+def _update_lims(pos, ax):
```
Comment (nwlandry) [inline xgi/drawing/draw.py:963]: Genius!

**Review by nwlandry — COMMENTED** [review]
This looks good, thanks! A few things:
* I would be in favor of keeping it as `net` instead of `H` in nodestats, but if you feel strongly, make sure that it's consistent.
* If you could add a test to check that the nodes, lines, edges are plotted on the right layer, that would be awesome, but frankly what you have is a huge improvement, so no sweat if not.
* I think that someone mentioned that the clustering coefficient is not between 0 and 1. Can you add this as an issue?
* I like adding the optional arguments to the draw nodes/edges functions.

Overall looks great and I think I'm good with merging this!

File: tests/drawing/test_draw.py (hunk 1 above)
Comment (maximelucas) [reply tests/drawing/test_draw.py:113]: Done

**Comment by maximelucas** [comment]
All done. I had already raised issue #309 about the clustering coefficient :) We'll need to test it once the right answer is decided.

**Pull request merged**